src/serializationStrategies/XpathRangeSelector/xpath.spec.js

```diff
@@ -179,33 +182,42 @@ describe('getXPathForElement', () => {
         <div></div>
         <div></div>
         <span></span>
-        <div><div id="target1"><span id="target2" ${DATA_ATTR}><span id="target3" ${DATA_ATTR}>asdfasdf</span></span></div></div>
-        <div></div>
+        <div>` +
+          `<div id="target1">` +
+            `<span id="target2" ${DATA_ATTR}>` +
+              screenReaderNode +
+              `<span id="target3" ${DATA_ATTR}>` +
+                `${screenReaderNode}asdfasdf${screenReaderNode}` +
+              `</span>` +
+              screenReaderNode +
+            `</span>` +
+          `</div>` +
+        `</div>` +
+        `<div></div>
       </div>
     `;
 
     const reference = document.getElementById('reference');
     const expectedPath = "./*[name()='div'][3]/*[name()='div'][1]";
-    const expectedOffset = 1;
 
     const target1 = document.getElementById('target1');
     const [result1, offset1] = xpath.getXPathForElement(target1, 1, reference);
     expect(result1).toEqual(expectedPath);
-    expect(offset1).toEqual(expectedOffset);
+    expect(offset1).toEqual(1);
```

Comment:
I don't know if changes to this test are correct

Reply:
I'm struggling with debugging this

Reply:
@TomWoodward  Could you please help me with establishing what is expected offset?
I think it should be still `1` for all test cases but if I understand correctly I have to change the second argument in `getXPathForElement` to adjust it for new nodes?
I've tried it and I'm always getting the wrong path or offset, so I have to probably do some work in `xpath.ts`, but until I'm not sure that I've correctly set tests it is hard to debug without knowing how this xpath stuff works

Reply:
yea its rough. what we're testing here is if our custom xpath logic correctly ignores the highlight elements, so the tests input a certain point and assert that it matches the expected output having ignored the extra nodes. because you're only adding highlight elements, i wouldn't expect any of the outputs to be different, but you will have to compensate your inputs in order to refer to the same place because you're adding nodes to the document.

resolving these offsets in my head takes more focus than i have available today, i will try to take a look tomorrow and help debug if it is the tests or the code that needs work.